OutOfSchool/OutOfSchool.DataAccess/Repository/ExternalImageStorage.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using MongoDB.Bson;
+using MongoDB.Driver.GridFS;
+using OutOfSchool.Services.Common.Exceptions;
+using OutOfSchool.Services.Contexts;
+
+namespace OutOfSchool.Services.Repository
+{
+    public class ExternalImageStorage : IExternalImageStorage
+    {
+        private readonly IGridFSBucket gridFsBucket;
+
+        public ExternalImageStorage(MongoDb db)
+        {
+            gridFsBucket = db.GetContext();
+        }
+
+        public async Task<Stream> GetByIdAsync(string pictureId)
+        {
+            try
+            {
+                return await gridFsBucket.OpenDownloadStreamAsync(new ObjectId(pictureId));
+            }
+            catch (Exception ex)
+            {
+                throw new ImageStorageException(ex);
+            }
+        }
+
+        public async Task<string> UploadImageAsync(Stream contentStream, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                contentStream.Position = uint.MinValue;
+                var objectId = await gridFsBucket.UploadFromStreamAsync(Guid.NewGuid().ToString(), contentStream, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
probably not relevant if we move to SQL storage, but maybe leave a comment so we don't forget. If we stick to Mongo then using objectId might not be cool as it's technically not an uuid and they are predictable. We can use the file name as identifier (and it's possible to search by filename) as it's almost unique :)